lib/src/command_processor.dart

```diff
@@ -59,4 +64,19 @@ class _CommandProcessor {
     request.headers.add(HttpHeaders.ACCEPT_CHARSET, UTF8.name);
     request.headers.add(HttpHeaders.CACHE_CONTROL, "no-cache");
   }
+
+  Future _getLock() async {
```

Comment:
This locking mechanism is great! Perhaps move it into it's own class (`Lock`) to make it re-usable?